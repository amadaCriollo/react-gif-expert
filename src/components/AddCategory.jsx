import { useState } from "react"

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState();

    const onInputChange = ( { target } ) => {
       setInputValue(target.value);
    }

    const onSubmit = ( event ) => {
        event.preventDefault();

        if(inputValue.trim().length <= 1) return;
        
        //Primera forma
        //setCategories(cat => [inputValue, ...cat  ] );

        //Segunda forma
        setInputValue('');
        onNewCategory( inputValue.trim() );
                
    }

  return (
    <form onSubmit={ onSubmit }>
        <input
            type="text"
            placeholder="Buscar gifs"
            value={inputValue}
            onChange={ onInputChange }        
        />
    </form>
  )
}

