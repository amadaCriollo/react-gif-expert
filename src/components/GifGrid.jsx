import { GiftItem } from './GiftItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';


export const GifGrid = ( { category, deleteCategory } ) => {

    const {images, isLoading} = useFetchGifs( category );
  
    return (
        <>  
            <h1>{ category }</h1>
            <button onClick={ ( event ) => deleteCategory( event ) } value={ category }>Eliminar</button>
            
            {
                isLoading && ( <h2 >Cargando...</h2>)
            }
           
            <div className='card-grid'>
                {
                     images.map( ( image )  => (
                        <GiftItem 
                            key={ image.id }
                            { ...image }
                        />
                     ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

