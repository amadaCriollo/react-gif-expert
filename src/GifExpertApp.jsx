import { useState } from 'react';
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

  

   const [ categories, setCategories ] = useState( ['One Punch'] );

   const onAddCategory = ( newCategory ) => {
      //Validar si ya contiene un elemento con el mismo nombre
      if(categories.includes(newCategory) )  return;

      //Una forma
        setCategories([ newCategory, ...categories ]);
        //Segunda forma
        //setCategories(cat => [...cat, 'Valorant' ]);

   }

   const onDeleteCategory = ( { target } ) => {
      setCategories(categories.filter(cat => target.value !== cat))
   }

  return (  
    <>
    {/* Titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory 
          //setCategories={ setCategories }
          onNewCategory={ onAddCategory }
      />
  
        { 
          categories.map( ( category ) => ( 
              
              <GifGrid 
                    key={ category }
                    category={ category } 
                    deleteCategory={ onDeleteCategory }
              />
             
          ))
          
        }
    </>
  );
}
