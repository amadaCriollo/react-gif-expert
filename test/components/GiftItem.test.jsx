import { render, screen } from "@testing-library/react";
import { GiftItem } from "../../src/components";


describe('Pruebas en GifItem', () => {
    
    const title = 'One Punch';
    const url = 'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=b8569c3658it1cpzvqsuts9i8xwt8es7bz34w814w084d9oo&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('Debe hacer match con snapshot ', () => {
        const { container }  = render ( <GiftItem title= { title } url = {url}/> );
        expect(container).toMatchSnapshot();       
    })

    test('Debe de mostrar la imagen con el URL y ALT indicado ', () => {
        render ( <GiftItem title= { title } url = {url}/> );
        //screen.debug();

        //Primera Forma
        // expect( screen.getByRole('img').src ).toBe( url )
        // expect( screen.getByRole('img').alt ).toBe( title );
        
        //Segunda Forma
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( src );
        expect( alt ).toBe( alt );

    });

    test('Debe mostrar el tÃ­tulo del componente ', () => {
        render ( <GiftItem title= { title } url = {url}/> );
        expect( screen.getByText( title ) ).toBeTruthy();
    })
    
    
        
})
