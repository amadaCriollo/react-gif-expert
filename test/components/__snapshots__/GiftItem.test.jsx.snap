// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=b8569c3658it1cpzvqsuts9i8xwt8es7bz34w814w084d9oo&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
       
      One Punch
    </p>
  </div>
</div>
`;
